src/Stateless/StateConfiguration.cs

```diff
@@ -415,7 +415,7 @@ public StateConfiguration OnEntry(Action entryAction, string entryActionDescript
                 Enforce.ArgumentNotNull(entryAction, nameof(entryAction));
                 return OnEntry(
                     t => entryAction(),
-                    entryActionDescription ?? entryAction.TryGetMethodName());
+                    entryActionDescription);
```

Comment:
Because the action is wrapped by the lambda expression `t => ...`, we'll lose the original method name with this change.